assessments/projects/cubefs/self-assessment.md

```diff
@@ -0,0 +1,300 @@
+
+ **Security Relevant** - 
+
+  * **Multiple Replicas**
+   CubeFS uses multiple replicas to meet multi-tenant requirements. The data between replicas is mirrored, and strong consistency replication protocols are utilized to ensure data consistency between different replicas. Users can flexibly configure different numbers of replicas according to their application scenarios. Though not a strategy whose main explicit purpose is security, there are nonetheless security benefits to it. By the redundancy of the design, the security of CubeFS is elevated as even if one replica becomes corrupted or unavailable by a fault or attack, by the user’s configuration there can be several backups of the same data to ensure performance is not compromised. As well, consider the following scenarios:
+
+    * In the case of a DDoS attack, where an attacker floods a system with a massive volume of requests to overwhelm and disrupt services, having multiple replicas helps distribute the incoming traffic. Load balancers can redirect requests across replicas, preventing a single point of failure and making it more challenging for attackers to overload a specific server.
+
+    * Multiple replicas provide protection against attacks that aim to compromise data integrity. If one replica is targeted by a data manipulation attack, the other replicas can act as a reference to identify and rectify unauthorized changes, preserving the integrity of the data.
+
+## Project compliance
+
+CubeFS does not currently document meeting particular compliance standards such as PCI-DSS, COBIT, ISO, GDPR, etc.
+
+## Secure development practices
+
+* **Development Pipeline** - 
+
+  * **Version control** - The CubeFS repository uses Git as its primary version control tool.
+
+  * **Code contribution** - CubeFS has defined a document for users and other community members who want to contribute to the repository. The document contains development style guidelines and commit message guidelines. 
+
+  * **Commit Signing** - All commits have to be signed following certain guidelines and go through a DCO (Developer Certificate of Origin) check. 
+
+  * **CI/CD pipeline** - CubeFS uses automated CI/CD pipelines to build and test code and uses the Travis CI service to run all checks including unit and integration tests. 

```

Comment:
Are there any SAST tools for container image vulnerabilities, static code analysis or source composition analysis included in their CI pipeline? Please include here.

Reply:
Lack of SAST tools in CubeFS CI/CD pipeline has been addressed.